Code/client/ServerRequestBuilder.cs

```diff
@@ -0,0 +1,905 @@
+
+                case "get detailed feedback for the item":
+                case "10":
+                    return BuildGetDetailedFeedbackRequest();
+
+                case "logout":
+                    return BuildLogoutrequest();
+
+                default:
+                    Console.WriteLine("Unknown command.");
+                    return null;
+            }
+        }
+
+        private string BuildLoginRequest()
+        {
+            string role = null;
+            Console.WriteLine("Which role would you like to log in as? (Employee, Admin, Chef)");
+
+            while (true)
+            {
+                Console.Write("Enter your role: ");
+                role = Console.ReadLine()?.Trim().ToLower();
+
+                if (roles.Contains(role.ToLower()))
+                {
+                    break; 
+                }
+                else
+                {
+                    Console.WriteLine("Invalid role. Please enter either 'Employee', 'Admin', or 'Chef'.");
+                }
+            }
+
+            UserLoginDTO user = new();
+            user.Role = role;
+
+            switch (role.ToLower())
+            {
+                case "employee":
+                    while(string.IsNullOrWhiteSpace(user.UserName))
+                    {
+                        Console.WriteLine("Enter your Employee Code:");
+                        user.UserName = Console.ReadLine();
+
+                        ValidateInputString(user.UserName,"employee code");
+                    }
+                    break;
+
+                default:
+                    while (string.IsNullOrWhiteSpace(user.UserName))
+                    {
+                        Console.WriteLine("Enter your UserName:");
+                        user.UserName = Console.ReadLine();
+
+                        ValidateInputString(user.UserName, "username");
+                    }
+                    break;
+            }
+
+            while (string.IsNullOrWhiteSpace(user.Password))
+            {
+                Console.WriteLine("Enter your Password:");
+                user.Password = Console.ReadLine();
+
+                ValidateInputString(user.Password, "password");
+            }
+
+            string userJson = JsonSerializer.Serialize(user);
+            return $"login#{userJson}";
+        }
+
+        private static string BuildLogoutrequest()
+        {
+            return $"logout#{UserData.UserId}";
+        }
+
+        private string BuildRegisterRequest()
+        {
+            string role = null;
+
+            Console.WriteLine("Which role would you like to log in as? (Employee, Admin, Chef)");
+            while (true)
+            {
+                while (string.IsNullOrWhiteSpace(role))
+                {
+                    Console.Write("Enter your role: ");
+                    role = Console.ReadLine()?.Trim().ToLower();
+                }
+
+                if (role == "employee" || role == "admin" || role == "chef")
```

Comment:
Instead of hardcoding the roles, we can have constants declared and use them. It makes it easier to maintain in case values need to be changed.